bootstrap/kubeadm/types/v1beta2/doc.go

```diff
@@ -14,263 +14,13 @@ See the License for the specific language governing permissions and
-// 	    -  "ec2-10-100-0-1.compute-1.amazonaws.com"
-// 	    peerCertSANs:
-// 	    - "10.100.0.1"
-// 	  # external:
-// 	    # endpoints:
-// 	    # - "10.100.0.1:2379"
-// 	    # - "10.100.0.2:2379"
-// 	    # caFile: "/etcd/kubernetes/pki/etcd/etcd-ca.crt"
-// 	    # certFile: "/etcd/kubernetes/pki/etcd/etcd.crt"
-// 	    # keyFile: "/etcd/kubernetes/pki/etcd/etcd.key"
-// 	networking:
-// 	  serviceSubnet: "10.96.0.0/12"
-// 	  podSubnet: "10.100.0.1/24"
-// 	  dnsDomain: "cluster.local"
-// 	kubernetesVersion: "v1.12.0"
-// 	controlPlaneEndpoint: "10.100.0.1:6443"
-// 	apiServer:
-// 	  extraArgs:
-// 	    authorization-mode: "Node,RBAC"
-// 	  extraVolumes:
-// 	  - name: "some-volume"
-// 	    hostPath: "/etc/some-path"
-// 	    mountPath: "/etc/some-pod-path"
-// 	    readOnly: false
-// 	    pathType: File
-// 	  certSANs:
-// 	  - "10.100.1.1"
-// 	  - "ec2-10-100-0-1.compute-1.amazonaws.com"
-// 	  timeoutForControlPlane: 4m0s
-// 	controllerManager:
-// 	  extraArgs:
-// 	    "node-cidr-mask-size": "20"
-// 	  extraVolumes:
-// 	  - name: "some-volume"
-// 	    hostPath: "/etc/some-path"
-// 	    mountPath: "/etc/some-pod-path"
-// 	    readOnly: false
-// 	    pathType: File
-// 	scheduler:
-// 	  extraArgs:
-// 	    address: "10.100.0.1"
-// 	extraVolumes:
-// 	- name: "some-volume"
-// 	  hostPath: "/etc/some-path"
-// 	  mountPath: "/etc/some-pod-path"
-// 	  readOnly: false
-// 	  pathType: File
-// 	certificatesDir: "/etc/kubernetes/pki"
-// 	imageRepository: "k8s.gcr.io"
-// 	useHyperKubeImage: false
-// 	clusterName: "example-cluster"
-// 	---
-// 	apiVersion: kubelet.config.k8s.io/v1beta1
-// 	kind: KubeletConfiguration
-// 	# kubelet specific options here
-// 	---
-// 	apiVersion: kubeproxy.config.k8s.io/v1alpha2
-// 	kind: KubeProxyConfiguration
-// 	# kube-proxy specific options here
-//
-// Kubeadm join configuration types
-//
-// When executing kubeadm join with the --config option, the JoinConfiguration type should be provided.
-//
-//    apiVersion: kubeadm.k8s.io/v1beta2
-//    kind: JoinConfiguration
-//       ...
-//
-// The JoinConfiguration type should be used to configure runtime settings, that in case of kubeadm join
-// are the discovery method used for accessing the cluster info and all the setting which are specific
-// to the node where kubeadm is executed, including:
-//
-// - NodeRegistration, that holds fields that relate to registering the new node to the cluster;
-// use it to customize the node name, the CRI socket to use or any other settings that should apply to this
-// node only (e.g. the node ip).
-//
-// - APIEndpoint, that represents the endpoint of the instance of the API server to be eventually deployed on this node.
-//
-package v1beta2 // import "sigs.k8s.io/cluster-api/bootstrap/kubeadm/kubeadm/v1beta2"
+IMPORTANT: Do not change thus files!
+IMPORTANT: Those types are meant only for KubeadmConfig serialization/deserialization, and should not be used for other purposes.
```

Comment:
```suggestion
IMPORTANT: Thge types in this package are meant only for KubeadmConfig serialization/deserialization, and should not be used for other purposes.
```

Can we make these `internal` ?

Reply:
I have considered this, but it turned out that it is tricky, because those v1beta1 types are still used in the v1alpha3 API.
So I have decided to keep this as it is now both for v1beta1 and v1beta2 for consistency.

What about if I open an issue to track this cleanup for whenever we are going to remove v1alpha3?

Reply:
Sounds good